pennylane/plugins/default_qubit.py

```diff
@@ -302,11 +302,11 @@ def apply(self, operation, wires, par):
                 raise ValueError('State vector must be of length 2**wires.')
             return
         elif operation == 'BasisState':
+            n = len(par[0])
             # get computational basis state number
-            if not (set(par[0]) == {0, 1} or set(par[0]) == {0} or set(par[0]) == {1}):
-                raise ValueError("BasisState parameter must be an array of 0/1 integers.")
+            if not (set(par[0]) == {0, 1} or set(par[0]) == {0} or set(par[0]) == {1}) or (wires != None and wires != [] and n != len(wires)):
```

Comment:
This is hard to parse. Can we change it to have two clauses? Say one to do with pars and another to do with wires?
Then do `if not pars_valid or not wires_valid`:

Reply:
Agree. f6e5b4d9a94c7995457fc94cb9e05fb865c65291 should hopefully make improve the checking here. I noticed that the default qubit plugin can actually only do BasisState on all wires.